### PR TITLE
[No QA] Replace usages of useWindowDimensions with useResponsiveLayout in components folder

### PR DESCRIPTION
**Pull request by rayane-d**
cc @getusha 
### Details

https://github.com/Expensify/App/issues/30528#issuecomment-1868485991
### Fixed Issues

$ https://github.com/Expensify/App/issues/30528

### Tests

Perform general regression testing verifying that app layouts and styles are correct.

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A
### QA Steps

No QA
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/77965000/6b564907-ca2c-41d5-b5a1-220cd71c0ac1

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/77965000/c59f7004-b6f9-48ca-aeec-1fa989bb21f2

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/77965000/918067cc-856b-447c-aacd-69cc32a6fe57

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/77965000/684b2af2-10d7-4964-8658-695d0abb7cf9

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/77965000/0a7f5b44-dc07-49d3-863c-c5cc715bf0df

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/77965000/296ef5de-e7a4-4105-ac20-413f328bf4cb

</details>

### CODE REVIEW

**Comment by getusha** [comment]
@rayane-djouah looks like we have a merge conflict.

File: src/pages/signin/SignInPage.js [hunk 1]
```diff
@@ -256,23 +251,21 @@ function SignInPageInner({credentials, account, isInModal, activeClients, prefer
         // Bottom SafeAreaView is removed so that login screen svg displays correctly on mobile.
         // The SVG should flow under the Home Indicator on iOS.
         <View
-            style={[styles.signInPage, StyleUtils.getSafeAreaPadding({...safeAreaInsets, bottom: 0, top: isInModal ? 0 : safeAreaInsets.top}, 1)]}
```
Comment (getusha) [inline src/pages/signin/SignInPage.js:259]: I think we should use `isInModal` here instead of `shouldUseNarrowLayout`

**Comment by getusha** [comment]
@rayane-djouah Do we need to add QA steps or mark it as No QA?

**Comment by getusha** [comment]
@rayane-djouah check my comments please.

**Comment by roryabraham** [comment]
Hmmm I think that sounds like it might be overkill in this case, but I'll make an AdHoc build

**Comment by roryabraham** [comment]
ah, sorry I think I was on mobile and couldn't kick it off. Doing that now...

**Comment by getusha** [comment]
Thanks @roryabraham, testing.

**Comment by getusha** [comment]
`BUG:` wrong animation when closing avatar modal

1. Open profile page
2. Press on the avatar
3. Close it.

We can notice it changes to small screen layout quickly

https://github.com/Expensify/App/assets/75031127/a84dfa0f-6dc8-4afc-bd7b-c68a7ddafcc2

**Comment by getusha** [comment]
`BUG:` Emoji picker strangely expanding from status page

1. Account settings > Profile > Status
2. Open the emoji picker from the dropdown

https://github.com/Expensify/App/assets/75031127/86f21444-e59e-4a8b-b4a3-185bdaf8e538

**Comment by getusha** [comment]
`BUG:` Wrong layout on the state picker page

1. Go to Account settings > Profile
2. Press on Address
3. State

https://github.com/Expensify/App/assets/75031127/42e11c15-fb66-4e14-b627-c91b9f2adc6f

**Comment by getusha** [comment]
Related? refreshing leads to not found page.

https://github.com/Expensify/App/assets/75031127/2e4a25ef-4bfd-4d1b-a8b0-818cd57c6137

**Comment by getusha** [comment]
@rayane-djouah Friendly bump

**Comment by getusha** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by getusha** [comment]
@rayane-djouah there are several places we shouldn't be showing the offline indicator

example:

Screenshot 2024-03-06 at 1 18 33 in the afternoon

File: src/components/ScreenWrapper.tsx [hunk 2]
```diff
@@ -262,8 +264,8 @@ function ScreenWrapper(
                                               })
                                             : children
                                     }
-                                    {isSmallScreenWidth && shouldShowOfflineIndicator && <OfflineIndicator style={offlineIndicatorStyle} />}
-                                    {!isSmallScreenWidth && shouldShowOfflineIndicatorInWideScreen && (
+                                    {shouldUseNarrowLayout && shouldShowOfflineIndicator && <OfflineIndicator style={offlineIndicatorStyle} />}
+                                    {!shouldUseNarrowLayout && shouldShowOfflineIndicatorInWideScreen && (
```
Comment (getusha) [inline src/components/ScreenWrapper.tsx:268]: we should use `isSmallScreenWidth` here to fix the ^ issue.

File: src/components/Section/index.tsx [hunk 3]
```diff
@@ -119,7 +119,7 @@ function Section({
                     {overlayContent?.()}
                 </View>
             )}
-            <View style={[styles.w100, isCentralPane && (isSmallScreenWidth ? styles.p5 : styles.p8)]}>
+            <View style={[styles.w100, isCentralPane && (shouldUseNarrowLayout ? styles.p5 : styles.p8)]}>
```
Comment (getusha) [inline src/components/Section/index.tsx:122]:
```suggestion
            <View style={[styles.w100, isCentralPane && (isSmallScreenWidth ? styles.p5 : styles.p8)]}>
```

Causing

https://github.com/Expensify/App/assets/75031127/35b1143a-1b58-47e4-b8bb-c89b784538f5

**Comment by getusha** [comment]
@rayane-djouah one more issue

Go to Workspace Settings > Bank Account > Start over

You can also notice a flickering of the size on the update to USD modal

https://github.com/Expensify/App/assets/75031127/c6e4082c-6bcb-4ee7-b819-2031625aceb9

**Comment by getusha** [comment]
I think we should avoid using `shouldUseNarrowLayout` in all Modals.

**Comment by roryabraham** [comment]
> I think we should avoid using shouldUseNarrowLayout in Modal components

I'm confused by this ... wasn't it one of the primary use-cases where we'd want to use a narrow layout even if we have a wide screen?

**Comment by getusha** [comment]
>I'm confused by this ... isn't this one of the primary use-cases where we'd want to use a narrow layout even if we have a wide screen?

@roryabraham when a modal shows up for example while the RHP is open it'll show a layout on bigger screens that's supposed to be shown only on narrow layouts. check this https://github.com/Expensify/App/pull/36292#issuecomment-1985925977

**Comment by getusha** [comment]
@rayane-djouah can you pull main and make sure every component is updated & [this](https://github.com/Expensify/App/pull/36292#issuecomment-1987334312)? thanks!

**Review by getusha — COMMENTED** [review]
@rayane-djouah can we add some `Tests` steps? or add No QA? also lets complete Screenshots/Videos section.

**Comment by roryabraham** [comment]
also, we've got conflicts here

**Comment by getusha** [comment]
> @getusha I think that this can be [No QA], do you agree?

I think it is fine since we touched a lot of components.

**Comment by getusha** [comment]
Testing.. @rayane-djouah we've got another conflict

**Review by getusha — COMMENTED** [review]
Everything seems to be working as expected, reassure & lint are failing @rayane-djouah

**Comment by dubielzyk-expensify** [comment]
Are there any visual changes as part of this PR?

**Comment by dubielzyk-expensify** [comment]
It gets triggered now when editing a styles file just so we can stay on top of things 😄

**Review by getusha — APPROVED** [review]

**Review by roryabraham — DISMISSED** [review]
stoked that this is ready to go

File: src/hooks/useResponsiveLayout.ts [hunk 4]
```diff
@@ -1,27 +1,52 @@
...
+    isLargeScreenWidth: boolean;
+    isExtraSmallScreenWidth: boolean;
+    isSmallScreen: boolean;
 };
 
 /**
- * Hook to determine if we are on mobile devices or in the Modal Navigator
+ * Hook to determine if we are on mobile devices or in the Modal Navigator.
+ * Use "shouldUseNarrowLayout" for "on mobile or in RHP/LHP", "isSmallScreenWidth" for "on mobile", "isInModal" for "in RHP/LHP".
+ * Note: Don't use "shouldUseNarrowLayout" in popovers and "alert-style" modals.
```
Comment (roryabraham) [inline src/hooks/useResponsiveLayout.ts:19]: Did we consider adding an automatic check for the modal type, such that you can safely use `shouldUseNarrowLayout` in any modal?

**Comment by getusha** [comment]
@rayane-djouah i think we may have caused a performance issue

There is a noticeable lag when opening popovers/RHPs

https://github.com/Expensify/App/assets/75031127/dd9ed69b-6c10-48bc-9d97-5d734879b138

**Comment by getusha** [comment]
@rayane-djouah still able to reproduce, i see you're testing this on safari could you try it on chrome?

https://github.com/Expensify/App/assets/75031127/1ee1ad6d-d11b-466b-8e21-095b6bef0716

**Comment by getusha** [comment]
| Staging | This PR |
|--------|--------|
| <video src="https://github.com/Expensify/App/assets/75031127/a89b2d36-57b7-4333-82d6-1a55b778248a" /> | <video src="https://github.com/Expensify/App/assets/75031127/7911306f-fe28-4223-a1ec-abf8fbae1db0" /> |

@rayane-djouah it still feels like there is a difference can you compare it with staging? it's not much i think we can ignore it, if it is not possible to fix. @roryabraham wdyt?

**Comment by roryabraham** [comment]
Looks similar enough to me to ignore 👍🏼

**Review by getusha — APPROVED** [review]
lgtm!

**Comment by roryabraham** [comment]
Merge freeze is still in effect for now

**Comment by roryabraham** [comment]
Merge freeze is lifted, I will try to review this today

**Review by roryabraham — APPROVED** [review]

**Comment by Christinadobrzyn** [comment]
hey @roryabraham and @getusha we might have a regression - can you check - https://github.com/Expensify/App/issues/42886#issuecomment-2141178027

**Pull request merged**